src/main/java/com/amazon/opendistroforelasticsearch/security/rest/TenantInfoAction.java

```diff
@@ -127,7 +136,46 @@ public void accept(RestChannel channel) throws Exception {
             }
         };
     }
-    
+
+    private boolean isAuthorized() {
+        final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+
+        if (user == null) {
+            return false;
+        }
+
+        // check if the user is a kibanauser or super admin
+        if (user.getName().equals(evaluator.kibanaServerUsername()) || adminDns.isAdmin(user)) {
+            return true;
+        }
+
+        // If user check failed by name and admin, check if the users belong to kibana opendistro role
+        final Tuple<Long, Settings.Builder> rolesMapping = load(ConfigConstants.CONFIGNAME_ROLES_MAPPING, true);
+        final Map<String, Object> rolesMappingConfiguration = Utils.convertJsonToxToStructuredMap(rolesMapping.v2().build());
+
+        // check if kibanaOpendistroRole is present in RolesMapping and if yes, check if user is a part of this role
+        if (rolesMappingConfiguration != null) {
+            String kibanaOpendistroRole = evaluator.kibanaOpendistroRole();
+            if (Strings.isNullOrEmpty(kibanaOpendistroRole)) {
+                return false;
+            }
+
+            Map<String, Object> kibanaOpendistroRoleEntry = (Map<String, Object>) rolesMappingConfiguration.getOrDefault(kibanaOpendistroRole, null);
+
+            if (kibanaOpendistroRoleEntry != null) {
+                ArrayList<String> kibanaOpendistroRoleUsers = (ArrayList<String>) kibanaOpendistroRoleEntry.get("users");
+                return kibanaOpendistroRoleUsers != null && kibanaOpendistroRoleUsers.contains(user.getName());
```

Comment:
I just now realized that this check is done only by username. The role mapping can be  based on backend role as well. Is there any other way to make this check based on both criterias.